controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -1170,6 +1212,17 @@ public void noteTimestampFromWriter(Controller.TimestampFromWriter request,
                 responseObserver, requestTag);
     }
 
+    @Override
+    public void deleteScopeRecursive(ScopeInfo request, StreamObserver<Controller.DeleteScopeRecursiveStatus> responseObserver) {
```

Comment:
When a deleteScopeRecursive is in progress a new CreateStream under the same Scope will return an internal Failure on the controller. This pattern repeats for other operations like CreateRG, CreateKVT, DeleteScope... We should try improving the return status so that the application receives the correct message.

Reply:
I've updated the issue #6446 to read the custom exception DeleteScopeInProgressException and return a proper message on client instead of controller failure.